apps/daimo-bot/api/PaymentActionProcessor.ts

```diff
@@ -0,0 +1,235 @@
+
+    this.trpcClient = trpc;
+    this.neynarClient = neynarClient;
+  }
+
+  async process() {
+    // Processes the incoming webhook event from Farcaster, extracting either a "request" or "pay" command,
+    // and performs actions based on the command type.
+
+    // 4 cases of Payment request:
+
+    // "@daimobot request":
+    // Case 1: Alice doesn't have FC linked ❌, requests $ from anyone (open-ended post)
+    //      Action 1: Daimobot responds with a link to register with Farcaster. Alice registers, then Daimobot responds with a link to request $
+    // Case 2: Alice has FC linked ✅, requests $ from anyone
+    //      Action 2: Daimobot responses with link that requests $ from anyone to Alice's Daimo address
+
+    // "@daimobot pay":
+    // Case 3: Alice responds to Bobs post to pay him, Bob doesn't have FC linked ❌
+    //     Action 3:  Daimobot responds with a link to register with Farcaster. Bob registers, then Daimobot responds with a link to request $
+    // Case 4: Alice responds to Bobs post to pay him, Bob has FC linked ✅
+    //     Action 4: Daimobot responds with link that requests $ from anyone to Bob's Daimo address
+
+    const daimobotCommand = this._tryExtractCommand();
+    if (!daimobotCommand) {
+      console.log("Cast follows neither request nor pay format.");
+      this.publishCastReply(
+        "Hi! To use me, please use the format `@daimobot request $<amount>` or `@daimobot pay $<amount>`."
+      );
+      return;
+    }
+
+    const { action, cleanedAmount } = daimobotCommand;
+    switch (action) {
+      case "request": {
+        // See if sender has Farcaster linked
+        console.log(
+          `[DAIMOBOT REQUEST] lookupEthereumAccountByFid for FID: ${this.senderFid}`
+        );
+        const senderEthAccount =
+          await this.trpcClient.lookupEthereumAccountByFid.query({
+            fid: this.senderFid,
+          });
+        if (!senderEthAccount) {
+          console.log(
+            "Sender not registered with Farcaster. Sending a response cast."
+          );
+          this.publishCastReply(
+            CONNECT_FC_MESSAGE
+            // TODO if there's a convenience url to connect Farcaster, add here
+          );
+          return;
+        }
+
+        const daimoShareUrl = await this.handleRequest(
+          cleanedAmount,
+          senderEthAccount
+        );
+        this.publishCastReply(
+          REQUEST_PAYMENT_MESSAGE(cleanedAmount, this.authorUsername),
+          {
+            embeds: [{ url: daimoShareUrl }],
+          }
+        );
+        break;
+      }
+      case "pay": {
+        // See if prospective recipient has Farcaster linked
+        if (!this.parentAuthorFid) {
+          console.warn(
+            "No parent author FID found, thus no one to prospectively pay."
+          );
+          this.publishCastReply(
+            "I can't find who you're trying to pay 🧐 \n\n To pay someone, make sure you're replying to one of their casts!"
```

Comment:
two small things:

1. i recommend factoring out `this.handleRequestCommand(...)` and `this.handlePayCommand(...)`
2. currently the english strings are split between here and botResponses.ts ; recommend picking one or other